### PR TITLE
Add missing stake->stake transition

### PR DESCRIPTION
**Pull request by clstrfsck**

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

**Review by LukasGasior1 — APPROVED** [review]
LGTM

File: src/main/java/com/radixdlt/atommodel/tokens/TokensConstraintScrypt.java [hunk 1]
```diff
@@ -492,6 +492,23 @@ private void defineStaking(SysCalls os) {
 			),
 			(in, meta) -> checkSignedBy(meta, in.getAddress())
 		));
+
+		// Stake movement
+		os.executeRoutine(new CreateFungibleTransitionRoutine<>(
```
Comment (talekhinezh) [inline src/main/java/com/radixdlt/atommodel/tokens/TokensConstraintScrypt.java:497]: I might be wrong, but I believe this allows someone to just move StakedTokens around to other users. Perhaps there should be a check that it's being sent to the same address.

Comment (clstrfsck) [reply src/main/java/com/radixdlt/atommodel/tokens/TokensConstraintScrypt.java:497]: OK, will add.

File: src/main/java/com/radixdlt/atommodel/tokens/TokensConstraintScrypt.java [hunk 2]
```diff
@@ -492,6 +492,26 @@ private void defineStaking(SysCalls os) {
...
+			checkEquals(
+				StakedTokensParticle::getGranularity,
+				StakedTokensParticle::getGranularity,
+				"Granularities not equal.",
+				StakedTokensParticle::getTokenPermissions,
+				StakedTokensParticle::getTokenPermissions,
+				"Permissions not equal.",
+				StakedTokensParticle::getAddress,
+				StakedTokensParticle::getAddress,
+				"Can't send staked tokens to another address."
```
Comment (clstrfsck) [inline src/main/java/com/radixdlt/atommodel/tokens/TokensConstraintScrypt.java:511]: Additional check added here.

File: src/test/java/com/radixdlt/atommodel/ParticlesEqualsHashCode.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		subTypes.addAll(findSubTypesInPkg("org.radix"));
+		subTypes.addAll(findSubTypesInPkg("com.radixdlt"));
+
+		subTypes.stream()
+			.filter(cls -> !Modifier.isAbstract(cls.getModifiers()))
+			.filter(cls -> !cls.equals(VoidParticle.class)) // Non-constructible
+			.forEachOrdered(this::testEquals);
+	}
+
+	private void testEquals(Class<? extends Particle> cls) {
```
Comment (clstrfsck) [inline src/test/java/com/radixdlt/atommodel/ParticlesEqualsHashCode.java:51]: InMemoryEngineStore required particles to have a working hashCode() and equals().

**Review by talekhinezh — APPROVED** [review]

**Review by siy — APPROVED** [review]
LGTM

**Pull request merged**